pansat/download/accounts.py

```diff
@@ -2,54 +2,192 @@
+from appdirs import user_config_dir
+from cryptography.fernet import Fernet
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.kdf.pbkdf2 import PBKDF2HMAC
+
+_IDENTITIES = {}
 """
-Dictionaries containing provider names and corresponding user names.
+Dictionary containing provider names and corresponding user names.
 """
 
 # The directory containing the configuration file.
-_app_dir = Path(user_config_dir("pansat", "pansat"))
-_app_dir.mkdir(parents=True, exist_ok=True)
+_APP_DIR = Path(user_config_dir("pansat", "pansat"))
+_APP_DIR.mkdir(parents=True, exist_ok=True)
 
 # The path to the configuration file.
-_identity_file = Path(_app_dir)
-_identity_file /= Path("identities.json")
+_IDENTITY_FILE = Path(_APP_DIR)
+_IDENTITY_FILE /= Path("identities.json")
+
+_PANSAT_SECRET = None
 
 ###############################################################################
 # Handling identities
 ###############################################################################
 
 
+def get_password(check=False):
+    """
+    Check if password is provided as ``PANSAT_PASSWORD`` variable.
+    If this is not the case query user to input password.
+
+    Params:
+        check(``bool``): Whether or user should insert the password twice
+            to avoid spelling errors.
+
+    Return:
+        ``str`` containing the password
+    """
+    try:
+        password = os.environ["PANSAT_PASSWORD"]
+    except KeyError:
+        print("Please enter your pansat user password:")
+        password = getpass.getpass()
+
+        if check:
+            print("Please repeat the password:")
+            password_re = getpass.getpass()
+            if password != password_re:
+                print("The two passwords don't match!")
+                get_password(check=check)
+
+    return password
+
+
+def hash_password(password, salt):
+    """
+    Computes the hash of a password.
+
+    Args:
+        password(``str``): The password
+        slat(``bytes``): The salt to use to compute the hash.
+
+    Return:
+        The base64 encoded hash of the password.
+    """
+    password = password.encode()
+    kdf = PBKDF2HMAC(algorithm=hashes.SHA256(), length=32, salt=salt, iterations=100000)
+    return base64.urlsafe_b64encode(kdf.derive(password))
+
+
+def decrypt(secret):
+    """
+    Decrypt password.
+
+    Args:
+        secret(``str``): The encrypted password
+
+    Return:
+        The decrypted password.
+
+    Raises:
+
+        cryptography.fernet.InvalidToken: Raised if the encrypted password
+            is invalid.
+    """
+    fernet = Fernet(_PANSAT_SECRET)
+    return fernet.decrypt(secret).decode("utf8")
+
+
+def encrypt(password):
+    """
+    Encript password.
+
+    Args:
+        password(``str``): The password to encrypt.
+
+    Return:
+        The encrypted password as ``bytes``.
+    """
+    fernet = Fernet(_PANSAT_SECRET)
+    return fernet.encrypt(password.encode())
+
+
+def authenticate():
```

Comment:
what if a user wants to use this in a script? Should there by a possibility to provide the pansat-password as an argument to the function? Then all the encrypting would of course be unnecessary..